servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/GrpcTelemetryStatus.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright © 2025 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.servicetalk.opentelemetry.http;
+
+import io.servicetalk.http.api.HttpHeaders;
+import io.servicetalk.http.api.HttpResponseMetaData;
+
+import javax.annotation.Nullable;
+
+final class GrpcTelemetryStatus {
+
+    private static final String GRPC_STATUS_HEADER_NAME = "grpc-status";
```

Comment:
We should use `CharSequence GRPC_STATUS = newAsciiString("grpc-status")` for more efficient internal comparison